### ISSUE TITLE
[Enhancement] Adding the X-adapter

### ISSUE THREAD

**Issue by George0726** [Mikubill/sd-webui-controlnet#2652]
Currently, X-adapter has released its codes and model. X-adapter makes the Controlnet for SD1.5 compatible with SDXL. It will be a great improvement on the current Diffusion community!
https://github.com/showlab/X-Adapter

**Comment by huchenlei** [comment]
The xadapter paper is potentially a scam according to my testing. Let's see what the author will respond. https://github.com/showlab/X-Adapter/issues/25

**Comment by huchenlei** [comment]
According to my testing in https://github.com/showlab/X-Adapter/issues/25, X-Adapter can be think as improved hiresfix, with lowres pass using SD15 checkpoint and highres pass using SDXL model. The adapter_guidance_start decides how much noise we are adding to the first pass result and how many steps to run in second pass.

Using second pass alone (Starting from pure random noise) won't give satisfying result. And this approach is highly dependent on SD15 model used. If a feature cannot be correctly interpreted by SD15 model, the SDXL model likely won't do a large modification on it.

Overall, I do not think it will worth the effort to implement X-Adapter.
